opensrp-giz-malawi/src/main/java/org/smartregister/giz/model/AdvancedSearchModel.java

```diff
@@ -0,0 +1,186 @@
+
+        if (response == null || response.isFailure() || StringUtils.isBlank(response.payload())) {
+            return matrixCursor;
+        }
+
+        JSONArray jsonArray = getJsonArray(response);
+        if (jsonArray != null) {
+
+            List<JSONObject> jsonValues = new ArrayList<>();
+            for (int i = 0; i < jsonArray.length(); i++) {
+                jsonValues.add(getJsonObject(jsonArray, i));
+            }
+
+            sortValues(jsonValues);
+
+            for (JSONObject client : jsonValues) {
+
+                if (client == null) {
+                    continue;
+                }
+
+                Map<String, String> childDetails = createChildDetailsMap();
+                CheckChildDetailsModel checkChildDetails = new CheckChildDetailsModel(client, childDetails).invoke();
+                if (checkChildDetails.is())
+                    continue;
+                String entityId = checkChildDetails.getEntityId();
+                String firstName = checkChildDetails.getFirstName();
+                String middleName = checkChildDetails.getMiddleName();
+                String lastName = checkChildDetails.getLastName();
+                String gender = checkChildDetails.getGender();
+                String dob = checkChildDetails.getDob();
+                String zeirId = checkChildDetails.getZeirId();
+                String epiCardNumber = checkChildDetails.getEpiCardNumber();
+                String inactive = checkChildDetails.getInactive();
+                String lostToFollowUp = checkChildDetails.getLostToFollowUp();
+                String nfcCardId = checkChildDetails.getNfcCardId();
+
+
+                CheckMotherDetailsModel checkMotherDetails = new CheckMotherDetailsModel(client).invoke();
+                String motherBaseEntityId = checkMotherDetails.getMotherBaseEntityId();
+                String motherFirstName = checkMotherDetails.getMotherFirstName();
+                String motherLastName = checkMotherDetails.getMotherLastName();
+
+                matrixCursor.addRow(new Object[]{entityId, null, firstName, middleName, lastName, gender, dob, zeirId, epiCardNumber, nfcCardId, motherBaseEntityId, motherFirstName, motherLastName, inactive, lostToFollowUp});
+            }
+
+            return matrixCursor;
+        } else {
+            return matrixCursor;
+        }
+    }
+
+    @NotNull
+    private Map<String, String> createChildDetailsMap() {
+
+        Map<String, String> childDetails = new HashMap<>();
+        childDetails.put("entityId", "");
+        childDetails.put("firstName", "");
+        childDetails.put("middleName", "");
+        childDetails.put("lastName", "");
+        childDetails.put("gender", "");
+        childDetails.put("dob", "");
+        childDetails.put("zeirId", "");
+        childDetails.put("epiCardNumber", "");
+        childDetails.put("inactive", "");
+        childDetails.put("lostToFollowUp", "");
+        childDetails.put("nfcCardId", "");
```

Comment:
I think you also need to add the mother details, which are retrieved from the `CheckMotherModelDetails`

Reply:
Fixed